compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -1687,4 +1687,23 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             false
         }
     }
+
+    pub(crate) fn is_field_suggestable(
+        &self,
+        field: &ty::FieldDef,
+        hir_id: HirId,
+        span: Span,
+    ) -> bool {
+        // The field must be visible in the containing module.
+        field.vis.is_accessible_from(self.tcx.parent_module(hir_id), self.tcx)
```

Comment:
I've taken `is_accessible_from(parent_module())` from one of the places I've removed but another one tries to obtain the `DefScope` instead of the parent module (via `adjust_ident_and_get_scope`). Which one is more correct / general?

Reply:
I think the latter, but I doubt it matters. If something is reachable from an item then it's reachable from its parent module, no?

Reply:
I wonder if there are some edge cases like `structs`s defined inside of a `fn` instead of a module. Not sure.